aiqleads/core/project_tracking.py

```
@@ -133,9 +204,23 @@
             return False
             
-        # TODO: Add more thorough validation against project_structure
+        # Check for directory traversal
+        if '..' in normalized_path:
+            return False
+            
+        # Ensure path matches project structure
+        allowed_dirs = ['core', 'utils', 'scripts', 'data', 'tests']
+        path_parts = Path(path).parts
+        if len(path_parts) > 1:
+            if path_parts[1] not in allowed_dirs:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)